src/font-stylesheet-timeout.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isDocumentReady} from './document-ready';
+import {timerFor} from './timer';
+
+/**
+ * While browsers put a timeout on font downloads (3s by default,
+ * some less on slow connections), there is no such timeout for style
+ * sheets. In the case of AMP external stylesheets are ONLY used to
+ * download fonts, but browsers have no reasonable timeout for
+ * stylesheets. Users may thus wait a long time for these to download
+ * even though all they do is reference fonts.
+ *
+ * For that reasons this function identifies (or rather infers) font
+ * stylesheets that have not downloaded within 1 second of the page
+ * response starting and reinserts equivalent link tags  dynamically. This
+ * removes their page-render-blocking nature and lets the doc render.
+ *
+ * 1 second was picked, because the font-stylesheets are typically
```

Comment:
Can we get away with a much shorter timeout? Considering font styles-sheets are pure declarations and should not have visual side-effects, even possibly blocking rendering for a second seems wasteful. Maybe a short timeout (<100 ms?) to avoid doing extra work in the case where css is cached and document becomes ready quickly but otherwise we lazy load the css.